# 0x05. Python - Exceptions

## Learning objectives

The following were the objectives of this project:
  * Identifying the difference between errors and exceptions
  * When to use exceptions
  * How to correctly handle an exception
  * Learning the purpose of catching exceptions
  * How to raise a builtin exception
  * When to implement a clean-up action after an exception

## Prototype functions :floppy_disk:

|       File           |             Prototype function              |  
|----------------------|---------------------------------------------|
|`0-safe_print_list.py`| `def safe_print_list(my_list=[], x=0):`     |
|`1-safe_print_integer.py`|`def safe_print_integer(value):`          |
|`2-safe_print_list_integers.py`| `def safe_print_list_integers(my_list=[], x=0`|
|`3-safe_print_division.py`| `def safe_print_division.py(a,b):`|
|`4-list_division.py`| `def list_division(my_list_1, my_list_2, list_length:`|
|`5-raise_exception.py`|`def raise_exception():`|
|`6-raise_exception_msg.py`|`def raise_exception_msg(message=''):`|
|`100-safe_print_integer_err.py`| `def safe_print_integer_err(value):`|
|`101-safe_function.py`|`def safe_function(fct, *args)`|
